### PR TITLE
Added AAB support for uploading to the Huawei AppGallery.

### PR DESCRIPTION
**Pull request by AesSedai101**

### CODE REVIEW

**Comment by pawelcala** [comment]
Thanks for collaboration. I will review PR tomorrow morning and try to deploy to bitrise repo asap.

File: step.sh [hunk 1]
```diff
@@ -30,7 +31,7 @@ printf "Obtaining a Token - Done\n"
 
 printf "\nObtaining the File Upload URL\n"
 curl --silent -X GET \
-  'https://connect-api.cloud.huawei.com/api/publish/v2/upload-url?appid='"${huawei_app_id}"'&suffix=apk' \
+  'https://connect-api.cloud.huawei.com/api/publish/v2/upload-url?appid='"${huawei_app_id}"'&suffix='"${FILE_EXT}" \
```
Comment (klamborowski) [inline step.sh:34]: @pawelcala 

The last apostrophe should be after FILE_EXT var like below, right?

`'https://connect-api.cloud.huawei.com/api/publish/v2/upload-url?appid='"${huawei_app_id}"'&suffix="${FILE_EXT}"'`

cc: @AesSedai101 did you successfully perform test run ?

Comment (AesSedai101) [reply step.sh:34]: @klamborowski yes, I currently have this running in a bitrise flow

File: step.yml [hunk 2]
```diff
@@ -66,16 +66,16 @@ inputs:
   - apk_path: $BITRISE_APK_PATH
```
Comment (klamborowski) [inline step.yml]: `apk_path `should be renamed to `file_path`

File: step.sh [hunk 3]
```diff
@@ -9,10 +9,11 @@ fi
 #Setup env vars
 LANG="${lang}"
 FILENAME_TO_UPLOAD="${huawei_filename}"
+FILE_EXT="${apk_path##*.}"
```
Comment (klamborowski) [inline step.sh]: `apk_path `should be renamed to `file_path`

File: step.sh [hunk 4]
```diff
@@ -9,10 +9,11 @@ fi
...
 
 printf "Apk path is: ${apk_path}\n"
```
Comment (klamborowski) [inline step.sh]: `apk_path `should be renamed to `file_path` and log message should be **File path** instead of **Apk path**

File: step.sh [hunk 5]
```diff
@@ -55,7 +56,6 @@ FILE_SIZE=`jq -r '.result.UploadFileRsp.fileInfoList[0].size' uploadfile.json`
```
Comment (klamborowski) [inline step.sh:56]: `apk_path` should be renamed to `file_path` also on line 52

obraz

**Review by klamborowski — CHANGES_REQUESTED** [review]

File: step.sh (hunk 5 above)
Comment (AesSedai101) [reply step.sh:56]: done

File: bitrise.yml [hunk 6]
```diff
@@ -3,7 +3,7 @@ default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
 
 app:
   envs:
-  - APK_PATH: ${APK_PATH}
+  - APK_PATH: ${FILE_PATH}
```
Comment (AesSedai101) [inline bitrise.yml]:
```suggestion
  - FILE_PATH: ${FILE_PATH}
```

**Review by klamborowski — APPROVED** [review]

**Comment by klamborowski** [comment]
@AesSedai101 FYI: https://github.com/bitrise-io/bitrise-steplib/pull/2602

**Pull request merged**